mapper/tedge_mapper/src/sm_c8y_mapper/mapper.rs

```diff
@@ -269,6 +278,41 @@ impl CumulocitySoftwareManagement {
         Ok(())
     }
 
+    async fn publish_restart_operation_status(
```

Comment:
This seems to be a lot of boilerplate code.

Reply:
Wouldn't it be better to abstract it for all the operations?. Maybe the same code is needed for other operations as well right?